phplib/models/Abbreviation.php

```diff
@@ -8,4 +8,21 @@ static function countAvailable($sourceId) {
       ->where('sourceID', $sourceId)
       ->count();
   }
+  
+  /**
+   * Returns, with constraints, first find abbreviation of form $short
+   * 
+   * @param   string  $excludedId all others
+   * @param   string  $short      abbreviation short form
+   * @param   int     $sourceId   source to search
+   * @return  ORMWrapper
+   */
+  static function getDuplicate($excludedId, $short, $sourceId) {
+    return Model::factory('Abbreviation')
+        ->where_raw('short = binary ?', $short)
```

Comment:
Dacă viteza nu e supărătoare, putem lăsa linia asta așa cum este.

Dacă există _lag_ prin Ajax și e supărător, putem schimba câmpul în `collate utf8_bin` ca să țină cont de literele mari/mici. Asta strică ordonarea alfabetică (căci se pierde informația că S ≠ Ș), dar putem face un câmp separat pentru ordonare (întâmplător tocmai am făcut asta la tabela `Tree`).

Reply:
Să vezi fază! Am făcut exact ce ai făcut tu, @CatalinFrancu . utf8-bin... și zic îi mai trag și un index unic pe `sourceId`+`short`, ca să nu-mi bat capul cu verificare prealabilă ci să prind excepția la salvare.  Dar în sinea mea: hm!, nu va fi de acord Cătă... și am refăcut așa cum e acum. Apoi am văzut ce ai făcut la `Tree`-uri. :) Fusăi pe drumul bun, dar nu e deloc grav prin metoda asta! Eu zic să rămână așa.